paasta_tools/kubernetes_tools.py

```diff
@@ -404,13 +404,22 @@ def __init__(self) -> None:
             ),
             fset=_set_disrupted_pods,
         )
+
         self.deployments = kube_client.AppsV1Api()
         self.core = kube_client.CoreV1Api()
         self.policy = kube_client.PolicyV1beta1Api()
         self.apiextensions = kube_client.ApiextensionsV1beta1Api()
         self.custom = kube_client.CustomObjectsApi()
         self.autoscaling = kube_client.AutoscalingV2beta2Api()
-        self.request = kube_client.ApiClient().request
+
+        self.api_client = kube_client.ApiClient()
+        self.request = self.api_client.request
+        # This function is used by the k8s client to serialize OpenAPI objects
+        # into JSON before posting to the api. The JSON output can be used
+        # in place of OpenAPI objects in client function calls. This allows us
+        # to monkey-patch the JSON data with configs the api supports, but the
+        # Python client lib may not yet.
+        self.serialize = self.api_client.sanitize_for_serialization
```

Comment:
So below in `get_autoscaling_metric_spec` we are calling `kube_client.serialize` to get the json version of the HPA object and returning that.  Am I correct in thinking that we need this here so that it will accept that already-serialized HPA object...?

Reply:
Or is this just for convenience so you can call `kube_client.serialize` instead of `kube_client.api_client.sanitize_for_serialization`?  If it's the latter I vote we not do this -- we're sort of doing a hacky thing, and I'd rather we showed that rather than making it look nicer.  (Also we're not really serializing -- that would result in a json string rather than a dict).

Reply:
It's a "make this hacky thing look nice" thing, but another reason I did this is because it makes it easier to document and reference how to do this in one place. I think it should be documented in some way because I suspect we'll have to do this from time to time because the Python client library is consistently behind the current k8s version, and we do do a fair amount of monkey-patching here and there.

If you have strong feelings about making it obvious a hacky thing is a hacky thing, then I'll remove teh `self.serialize` obfuscation.

Reply:
Hmm I guess that is fine, although I think it's not super likely to people will think to come find this comment when they just see `kube_client.serialize()` -- it looks pretty proper and official.  

Maybe you could either add a comment to the place where you use it or call it something more obvious like `convert_to_dict_representation` or something?  Your call

Reply:
I changed to `.jsonify` and added a comment that it is a hack.